packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -357,59 +392,85 @@ def get_atomic_swap_batch_transaction_proposal(
 
         return tx_message
 
-    def get_hash_single_transaction(self, terms) -> TransactionMessage:
+    def get_hash_single_transaction(
+        self,
+        from_address,
+        to_address,
+        item_id,
+        from_supply,
+        to_supply,
+        value,
+        trade_nonce,
+        ledger_api,
+        skill_callback_id,
+    ) -> TransactionMessage:
         """Sign the transaction before send them to agent1."""
         # assert self.address == terms.to_address
-        from_address_hash = self.instance.functions.getAddress(
-            terms.from_address
-        ).call()
-        to_address_hash = self.instance.functions.getAddress(terms.to_address).call()
+        from_address_hash = self.instance.functions.getAddress(from_address).call()
+        to_address_hash = self.instance.functions.getAddress(to_address).call()
+        value_eth_wei = ledger_api.api.toWei(value, "ether")
         tx_hash = Helpers().get_single_hash(
             _from=from_address_hash,
             _to=to_address_hash,
-            _id=terms.item_id,
-            _from_value=terms.from_supply,
-            _to_value=terms.to_supply,
-            _value_eth=terms.value_eth_wei,
-            _nonce=terms.trade_nonce,
+            _id=item_id,
+            _from_value=from_supply,
+            _to_value=to_supply,
+            _value_eth=value_eth_wei,
+            _nonce=trade_nonce,
         )
 
         tx_message = TransactionMessage(
             performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
-            skill_callback_ids=[ContractId("fetchai", "erc1155_skill", "0.1.0")],
-            tx_id="contract_deployment",
-            tx_sender_addr=terms.from_address,
+            skill_callback_ids=[skill_callback_id],
+            tx_id="contract-sign-hash",
+            tx_sender_addr=from_address,
             tx_counterparty_addr="",
             tx_amount_by_currency_id={"ETH": 0},
             tx_sender_fee=0,
             tx_counterparty_fee=0,
             tx_quantities_by_good_id={},
             info={},
             ledger_id="ethereum",
-            signing_payload={"tx_hashh": tx_hash},
+            signing_payload={"tx_hash": tx_hash},
         )
 
         return tx_message
 
-    def get_hash_batch_transaction(self, terms):
+    def get_hash_batch_transaction(
+        self,
+        from_address,
+        to_address,
+        item_ids,
+        from_supplies,
+        to_supplies,
+        value,
+        trade_nonce,
+        ledger_api,
+    ):
         """Sign the transaction before send them to agent1."""
         # assert self.address == terms.to_address
-        from_address_hash = self.instance.functions.getAddress(
-            terms.from_address
-        ).call()
-        to_address_hash = self.instance.functions.getAddress(terms.to_address).call()
+        from_address_hash = self.instance.functions.getAddress(from_address).call()
```

Comment:
Aren't we able to generate these hashes client side now without a contract call?

Reply:
No, cause the web3 hashes differently than the vyper. I will check again though.

Reply:
Check what we have in the research repo please, just to be sure.